### PR TITLE
Add option to list aliases

### PR DESCRIPTION
**Pull request by lrusnac**

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -77,6 +81,11 @@ func (c App) PrintUsage() {
 	c.flag.PrintDefaults()
 }
 
+// PrintAliasList prints, to the standard output, the list of aliases
+func (c App) PrintAliasList(aliases []string) {
+	fmt.Printf("alias list: %s\n", strings.Join(aliases, ", "))
+}
+
 // String returns a string representation of an App.
 func (c App) String() string {
```
Comment (davrodpin) [inline cli/cli.go]: Please add `aliases` to the returned string

File: cmd/mole/main.go [hunk 2]
```diff
@@ -47,7 +47,12 @@ func main() {
 		if err != nil {
 			os.Exit(1)
 		}
-
+	case "aliases":
+		aliases, err := lsAliases(*app)
```
Comment (davrodpin) [inline cmd/mole/main.go]: Check if the error is not nil before printing the aliases out.

File: storage/storage.go [hunk 3]
```diff
@@ -59,6 +59,20 @@ func FindByName(name string) (*Tunnel, error) {
 	return tun, nil
 }
 
+func FindAll() ([]string, error) {
```
Comment (davrodpin) [inline storage/storage.go]: This method should return a copy of `Store.Tunnels` and the caller function/method should filter what values it is interested on.

**Comment by lrusnac** [comment]
> Would you mind to rebase with master and squash your commits before I take a look again?
of course, should be fine now

**Comment by davrodpin** [comment]
@lrusnac, how about showing all attributes of an alias that is not empty or the default value, so users would know what they are supposed to do? Something like:

```
alias-example1
  local: 127.0.0.1:3306
  remote: 127.0.0.1:3306
  server: example@172.17.0.100

alias-example2
  remote: :3306
  server: my-database-server
  verbose: true
```

**Comment by lrusnac** [comment]
right, I feel like this looks a bit too verbose, don't you think? like if I have 10-20 aliases my standard output would be crazy.
there are multiple ways to go around that though:
- 1 liner, so put all the info of an alias in one line
- a verbose flag that would show details of all aliases like you suggested, and just names by default
- alias details, so getting this explicit output for 1 alias at a time that you specify

either way, this opens a bigger door in my opinion

**Comment by davrodpin** [comment]
> right, I feel like this looks a bit too verbose, don't you think? like if I have 10-20 aliases my standard output would be crazy.
> there are multiple ways to go around that though:
> 
> * 1 liner, so put all the info of an alias in one line
> * a verbose flag that would show details of all aliases like you suggested, and just names by default
> * alias details, so getting this explicit output for 1 alias at a time that you specify
> 
> either way, this opens a bigger door in my opinion

Once we have a way to list all aliases and their options, we can implement other options to perform specific tasks like showing the configuration of a specific alias and specify the output format (e.g. one-line, csv, json, xml, toml, etc.)

**Comment by lrusnac** [comment]
> Once we have a way to list all aliases and their options, we can implement other options to perform specific tasks like showing the configuration of a specific alias and specify the output format (e.g. one-line, csv, json, xml, toml, etc.)

agree, and this PR has a list of all the tunnels and prints out just the names, it can be further developed in what you are mentioning here

**Comment by davrodpin** [comment]
> agree, and this PR has a list of all the tunnels and prints out just the names, it can be further developed in what you are mentioning here

Well played 😄. Could you solve the conflict, so I can accept the PR?

**Comment by davrodpin** [comment]
Thanks!

**Comment by lrusnac** [comment]
:D np, will take a look for more options on print, a bit later in the week, cheers

**Pull request merged**